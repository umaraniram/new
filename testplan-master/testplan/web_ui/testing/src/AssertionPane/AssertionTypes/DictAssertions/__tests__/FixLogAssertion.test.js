import React from 'react';
import {shallow} from 'enzyme';
import {StyleSheetTestUtils} from "aphrodite";

import FixLogAssertion from '../FixLogAssertion';

function defaultProps() {
  return {
    assertion: {
      "category":"DEFAULT",
      "machine_time":"2019-02-12T17:41:43.314860+00:00",
      "description":null,
      "flattened_dict":[
        [0,555,""],
        [0,"",""],
        [1,624,["int","1"]],
        [1,556,["str","USD"]],
        [0,"",""],
        [1,624,["int","2"]],
        [1,556,["str","EUR"]],
        [0,36,["int","6"]],
        [0,38,["int","5"]],
        [0,22,["int","5"]],
        [0,55,["int","2"]]
      ],
      "line_no":688,
      "meta_type":"entry",
      "type":"FixLog",
      "utc_time":"2019-02-12T17:41:43.314849+00:00"
    }
  };
}


describe('FixLogAssertion', () => {
  let props;
  let shallowComponent;

  beforeEach(() => {
    // Stop Aphrodite from injecting styles, this crashes the tests.
    StyleSheetTestUtils.suppressStyleInjection();
    props = defaultProps();
    shallowComponent = undefined;
  });

  it('shallow renders the correct HTML structure', () => {
    shallowComponent = shallow(<FixLogAssertion {...props}/>);
    expect(shallowComponent).toMatchSnapshot();
  });
});