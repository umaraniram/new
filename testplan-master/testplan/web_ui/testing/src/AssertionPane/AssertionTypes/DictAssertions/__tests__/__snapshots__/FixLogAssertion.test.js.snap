// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FixLogAssertion shallow renders the correct HTML structure 1`] = `
<DictBaseAssertion
  buttonGroup={
    <DictButtonGroup
      flattenedDict={
        Array [
          Array [
            0,
            555,
            "",
          ],
          Array [
            0,
            "",
            "",
          ],
          Array [
            1,
            624,
            Array [
              "int",
              "1",
            ],
          ],
          Array [
            1,
            556,
            Array [
              "str",
              "USD",
            ],
          ],
          Array [
            0,
            "",
            "",
          ],
          Array [
            1,
            624,
            Array [
              "int",
              "2",
            ],
          ],
          Array [
            1,
            556,
            Array [
              "str",
              "EUR",
            ],
          ],
          Array [
            0,
            36,
            Array [
              "int",
              "6",
            ],
          ],
          Array [
            0,
            38,
            Array [
              "int",
              "5",
            ],
          ],
          Array [
            0,
            22,
            Array [
              "int",
              "5",
            ],
          ],
          Array [
            0,
            55,
            Array [
              "int",
              "2",
            ],
          ],
        ]
      }
      setRowData={[Function]}
      sortTypeList={
        Array [
          1,
          2,
        ]
      }
    />
  }
  columns={
    Array [
      Object {
        "cellStyle": Object {
          "border": "1px solid",
          "borderColor": "#d9dcde",
          "fontSize": "small",
          "padding": "6px 11px 6px 11px",
        },
        "field": "key",
        "headerStyle": Object {
          "border": "1px solid",
          "borderColor": "#d9dcde",
        },
        "render": [Function],
        "title": "Key",
      },
      Object {
        "cellStyle": Object {
          "backgroundColor": "#BDC3C750",
          "border": "1px solid",
          "borderColor": "#d9dcde",
          "fontSize": "small",
          "padding": "6px 11px 6px 11px",
        },
        "field": "value",
        "headerStyle": Object {
          "border": "1px solid",
          "borderColor": "#d9dcde",
        },
        "render": [Function],
        "title": "Value",
      },
    ]
  }
  rows={
    Array [
      Object {
        "descriptor": Object {
          "indent": 0,
          "isEmptyLine": false,
          "isListKey": true,
          "lineNo": 688,
          "status": undefined,
        },
        "key": Object {
          "type": "key",
          "value": 555,
        },
        "value": Object {
          "type": undefined,
          "value": undefined,
        },
      },
      Object {
        "descriptor": Object {
          "indent": 0,
          "isEmptyLine": true,
          "isListKey": false,
          "lineNo": 688,
          "status": undefined,
        },
        "key": Object {
          "type": null,
          "value": null,
        },
      },
      Object {
        "descriptor": Object {
          "indent": 1,
          "isEmptyLine": false,
          "isListKey": false,
          "lineNo": 688,
          "status": undefined,
        },
        "key": Object {
          "type": "key",
          "value": 624,
        },
        "value": Object {
          "type": "int",
          "value": "1",
        },
      },
      Object {
        "descriptor": Object {
          "indent": 1,
          "isEmptyLine": false,
          "isListKey": true,
          "lineNo": 688,
          "status": undefined,
        },
        "key": Object {
          "type": "key",
          "value": 556,
        },
        "value": Object {
          "type": "str",
          "value": "USD",
        },
      },
      Object {
        "descriptor": Object {
          "indent": 0,
          "isEmptyLine": true,
          "isListKey": false,
          "lineNo": 688,
          "status": undefined,
        },
        "key": Object {
          "type": null,
          "value": null,
        },
      },
      Object {
        "descriptor": Object {
          "indent": 1,
          "isEmptyLine": false,
          "isListKey": false,
          "lineNo": 688,
          "status": undefined,
        },
        "key": Object {
          "type": "key",
          "value": 624,
        },
        "value": Object {
          "type": "int",
          "value": "2",
        },
      },
      Object {
        "descriptor": Object {
          "indent": 1,
          "isEmptyLine": false,
          "isListKey": false,
          "lineNo": 688,
          "status": undefined,
        },
        "key": Object {
          "type": "key",
          "value": 556,
        },
        "value": Object {
          "type": "str",
          "value": "EUR",
        },
      },
      Object {
        "descriptor": Object {
          "indent": 0,
          "isEmptyLine": false,
          "isListKey": false,
          "lineNo": 688,
          "status": undefined,
        },
        "key": Object {
          "type": "key",
          "value": 36,
        },
        "value": Object {
          "type": "int",
          "value": "6",
        },
      },
      Object {
        "descriptor": Object {
          "indent": 0,
          "isEmptyLine": false,
          "isListKey": false,
          "lineNo": 688,
          "status": undefined,
        },
        "key": Object {
          "type": "key",
          "value": 38,
        },
        "value": Object {
          "type": "int",
          "value": "5",
        },
      },
      Object {
        "descriptor": Object {
          "indent": 0,
          "isEmptyLine": false,
          "isListKey": false,
          "lineNo": 688,
          "status": undefined,
        },
        "key": Object {
          "type": "key",
          "value": 22,
        },
        "value": Object {
          "type": "int",
          "value": "5",
        },
      },
      Object {
        "descriptor": Object {
          "indent": 0,
          "isEmptyLine": false,
          "isListKey": undefined,
          "lineNo": 688,
          "status": undefined,
        },
        "key": Object {
          "type": "key",
          "value": 55,
        },
        "value": Object {
          "type": "int",
          "value": "2",
        },
      },
    ]
  }
/>
`;
